solr/core/src/java/org/apache/solr/search/ValueSourceParser.java

```diff
@@ -975,6 +976,13 @@ public ValueSource parse(FunctionQParser fp) throws SyntaxError {
       }
     });
 
+    addParser("agg_uniqueBlockQuery", new ValueSourceParser() {
+      @Override
+      public ValueSource parse(FunctionQParser fp) throws SyntaxError {
```

Comment:
Hello, @a-siuniaev. Thanks for the patch. Wouldn't you mind to explore [Yonik's idea](https://issues.apache.org/jira/browse/SOLR-12325?focusedCommentId=16649083&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-16649083) about  overloading the same name with the different argument type (query)

Reply:
Added parseNamedArg to FunctionQParser.